provider/fs/src/manifest.rs

```diff
@@ -18,33 +18,23 @@ pub enum AliasOption {
 }
 
 #[non_exhaustive]
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
-pub enum SyntaxOption {
-    /// Serialize using JavaScript Object Notation (JSON).
-    Json,
-    /// Serialize using Bincode.
-    Bincode,
-    // Future: Consider adding a custom format option here.
-    // Custom {
-    //     file_extension: String,
-    // }
-}
-
-impl SyntaxOption {
-    /// Gets the file extension associated with the given syntax.
-    pub fn get_file_extension(&self) -> &str {
-        match self {
-            Self::Json => "json",
-            Self::Bincode => "bincode",
-        }
-    }
-}
-
-#[non_exhaustive]
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
 pub(crate) struct Manifest {
     /// Strategy for de-duplicating locale data.
     pub aliasing: AliasOption,
     /// Which data serialization file format is used.
-    pub syntax: SyntaxOption,
+    #[serde(rename = "syntax")]
+    pub buffer_format: BufferFormat,
+}
+
+impl Manifest {
+    /// Gets the file extension associated with the given buffer format in the manifest.
+    pub(crate) fn get_file_extension(&self) -> &str {
+        match self.buffer_format {
+            BufferFormat::Json => "json",
+            BufferFormat::Bincode1 => "bincode",
+            BufferFormat::Postcard07 => "postcard",
+            _ => "u",
```

Comment:
suggestion (nb): perhaps this should panic, or return an error or something?

(unsure if the error cases are actually possible)

Reply:
Commented